plugins/org.locationtech.udig.tool.edit/src/org/locationtech/udig/tools/edit/commands/SelectFeaturesAtPointCommand.java

```diff
@@ -193,6 +219,96 @@ private void runSelectionStrategies( IProgressMonitor monitor, FeatureIterator<S
                   false);
             }
         }
+        */
+        //code that enables selection of appropriate feature when multiple exist
+        //in the point where mouse click takes place
+        //******************************************************************/
+        List<SimpleFeature> featureList =new ArrayList<SimpleFeature>();
+        while (reader.hasNext()){
+            featureList.add(reader.next());
+        }      
+        final SimpleFeature[] features = featureList.toArray(new SimpleFeature[]{});
+
+        if (features.length == 1) {
+            for( SelectionStrategy selectionStrategy : strategies ) {
+                selectionStrategy.run(monitor, compositeCommand, parameters, features[0],
+                        true);
+            }
+        } else if (features.length > 1) {
+            PlatformGIS.syncInDisplayThread(new Runnable() {
+                @Override
+                public void run() {
+                    final Menu menu = new Menu(((ViewportPane) event.source).getControl().getShell(), SWT.POP_UP);
+                    for (final SimpleFeature feat : features) {
+                        MenuItem item = new MenuItem(menu, SWT.PUSH);
+                        //SimpleFeature feature=iter.next();
+                        boolean hasNameAttrib = feat.getAttribute(ATTRIBUTE_NAME) != null 
```

Comment:
I will have a look at it. However it will need to iterate all featureType attribute names and perform comparisons.